### PR TITLE
Upgrade Pinot Libraries to 1.1.0

### PR DESCRIPTION
**Pull request by wendigo**
## Description

Fixes https://github.com/trinodb/trino/issues/19919

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
@elonazoulay I've got it mostly working. Legacy (Non-grpc) is failing for 3 tests that I've disabled. Can you take a look?

**Comment by wendigo** [comment]
`[WARNING] Tests run: 534, Failures: 0, Errors: 0, Skipped: 176` 🥳

**Review by xiangfu0 — APPROVED** [review]

**Comment by wendigo** [comment]
@xiangfu0 unfortunately this change is not yet ready as I was forced to disable 3 failing tests that are returning incorrect results. Seems like a Pinot bug.

**Comment by xiangfu0** [comment]
> @xiangfu0 unfortunately this change is not yet ready as I was forced to disable 3 failing tests that are returning incorrect results. Seems like a Pinot bug.

Thanks, can you link the PRs here? I can help prioritize.

**Comment by wendigo** [comment]
@xiangfu0 I don't have fix yet, disabled tests are here: https://github.com/trinodb/trino/pull/21234/files#diff-b1808e48847579e72593f11e3128e594b8bf799690a48c1f3bb3319aeffd5231R46

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestingPinotCluster.java [hunk 1]
```diff
@@ -65,8 +65,8 @@
 public class TestingPinotCluster
         implements Closeable
 {
-    public static final String PINOT_LATEST_IMAGE_NAME = "apachepinot/pinot:0.12.1";
-    public static final String PINOT_PREVIOUS_IMAGE_NAME = "apachepinot/pinot:0.11.0";
+    public static final String PINOT_LATEST_IMAGE_NAME = "apachepinot/pinot:1.0.0";
+    public static final String PINOT_PREVIOUS_IMAGE_NAME = "apachepinot/pinot:0.12.1";
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestingPinotCluster.java:69]: Could you update the requirement in pinot.md?

File: plugin/trino-pinot/pom.xml [hunk 2]
```diff
@@ -14,15 +14,32 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <dep.pinot.version>0.12.1</dep.pinot.version>
+        <dep.pinot.version>1.1.0</dep.pinot.version>
+        <!--
+           Project's default for air.test.parallel is 'methods'. By design, 'instances' makes TestNG run tests from one class in a single thread.
+           As a side effect, it prevents TestNG from initializing multiple test instances upfront, which happens with 'methods'.
```
Comment (ebyhr) [inline plugin/trino-pinot/pom.xml:20]: This connector uses JUnit. Mentioning TestNG seems weird for me.

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotConnectorSmokeTest.java [hunk 3]
```diff
@@ -907,10 +907,10 @@ public void testShowCreateTable()
         assertThat((String) computeScalar("SHOW CREATE TABLE region"))
                 .isEqualTo(
                         "CREATE TABLE %s.%s.region (\n" +
-                                "   regionkey bigint,\n" +
-                                "   updated_at_seconds bigint,\n" +
+                                "   comment varchar,\n" +
                                 "   name varchar,\n" +
-                                "   comment varchar\n" +
+                                "   regionkey bigint,\n" +
+                                "   updated_at_seconds bigint\n" +
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotConnectorSmokeTest.java:913]: Do you know why the column order is changed? Is it possible to restore the original behavior?

Comment (wendigo) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/BasePinotConnectorSmokeTest.java:913]: Seems not possible. Columns are sorted

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java [hunk 4]
```diff
@@ -38,4 +40,25 @@ protected boolean isGrpcEnabled()
     {
         return false;
     }
+
+    @Test
+    @Override
+    public void testPassthroughQueriesWithAliases()
+    {
+        abort("Broken :( Elon help!");
```
Comment (elonazoulay) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java:48]: Ok :)

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestingPinotCluster.java (hunk 1 above)
Comment (elonazoulay) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestingPinotCluster.java:69]: Update: in a follow up pr, we can remove support for previous version. We spoke with Pinot committers and will rely on Pinot for backwards compatibility, going forward Pinot 1.x.x+ will manage compatibility. This also simplifies the code and removes some dependency issues.

**Comment by elonazoulay** [comment]
> @elonazoulay I've got it mostly working. Legacy (Non-grpc) is failing for 3 tests that I've disabled. Can you take a look?

Yessir!

**Comment by wendigo** [comment]
Locally tests are passing:
```
[INFO] Results:
[INFO]
[WARNING] Tests run: 539, Failures: 0, Errors: 0, Skipped: 175
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  06:33 min
[INFO] Finished at: 2024-05-24T09:12:34+02:00
[INFO] ------------------------------------------------------------------------
```

Great work @elonazoulay 🎉  This is the last outdated Trino plugin.

**Review by elonazoulay — APPROVED** [review]

File: plugin/trino-pinot/pom.xml [hunk 5]
```diff
@@ -14,15 +14,25 @@
...
+        <air.test.jvm.additional-arguments>${air.test.jvm.additional-arguments.default}
+            --add-exports=java.base/jdk.internal.ref=ALL-UNNAMED
+            --add-exports=java.base/sun.nio.ch=ALL-UNNAMED
+            --add-exports=jdk.unsupported/sun.misc=ALL-UNNAMED
+            --add-exports=jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED
+            --add-opens=jdk.compiler/com.sun.tools.javac=ALL-UNNAMED
+            --add-opens=java.base/java.lang=ALL-UNNAMED
+            --add-opens=java.base/java.lang.reflect=ALL-UNNAMED
+            --add-opens=java.base/java.io=ALL-UNNAMED
+            --add-opens=java.base/java.util=ALL-UNNAMED</air.test.jvm.additional-arguments>
```
Comment (ebyhr) [inline plugin/trino-pinot/pom.xml:27]: > @ebyhr Are these arguments required not only in tests but also in actual Trino clusters?
> @elonazoulay Yes

How did you confirm these JVM options are required in actual clusters? I ran `TestPinotConnectorSmokeTest` without additional JVM options and only testTimeBoundary is affected. The failure place is testing helper method (`createAndPopulateMixedCaseHybridTablesAndTopic`), not product path.

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java [hunk 6]
```diff
@@ -27,12 +25,6 @@ protected boolean isSecured()
         return false;
     }
 
-    @Override
-    protected String getPinotImageName()
```
Comment (ebyhr) [inline plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java:31]: The method in BasePinotConnectorSmokeTest is redundant now.

File: plugin/trino-pinot/pom.xml (hunk 5 above)
Comment (wendigo) [reply plugin/trino-pinot/pom.xml:27]: The failure is on the pinot-path:
```
	at net.openhft.chronicle.core.internal.ClassUtil.getSetAccessible0Method(ClassUtil.java:32)
	at net.openhft.chronicle.core.internal.ClassUtil.<clinit>(ClassUtil.java:13)
	at net.openhft.chronicle.core.Jvm.getField(Jvm.java:541)
	at net.openhft.chronicle.core.Jvm.maxDirectMemory0(Jvm.java:906)
	at net.openhft.chronicle.core.Jvm.<clinit>(Jvm.java:145)
	at org.apache.pinot.segment.spi.utils.JavaVersion.<clinit>(JavaVersion.java:32)
	at org.apache.pinot.segment.spi.memory.PinotDataBuffer.createDefaultFactory(PinotDataBuffer.java:169)
	at org.apache.pinot.segment.spi.memory.PinotDataBuffer.createDefaultFactory(PinotDataBuffer.java:162)
	at org.apache.pinot.segment.spi.memory.PinotDataBuffer.<clinit>(PinotDataBuffer.java:132)
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentDictionaryCreator.writeBytesValueDictionary(SegmentDictionaryCreator.java:268)
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentDictionaryCreator.build(SegmentDictionaryCreator.java:218)
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentColumnarIndexCreator.init(SegmentColumnarIndexCreator.java:184)
	at org.apache.pinot.segment.local.segment.creator.impl.SegmentIndexCreationDriverImpl.build(SegmentIndexCreationDriverImpl.java:228)
	at io.trino.plugin.pinot.BasePinotConnectorSmokeTest.createSegment(BasePinotConnectorSmokeTest.java:669)
	at io.trino.plugin.pinot.BasePinotConnectorSmokeTest.createAndPopulateMixedCaseHybridTablesAndTopic(BasePinotConnectorSmokeTest.java:469)
	at io.trino.plugin.pinot.BasePinotConnectorSmokeTest.createQueryRunner(BasePinotConnectorSmokeTest.java:156)
	at io.trino.testing.AbstractTestQueryFramework.init(AbstractTestQueryFramework.java:113)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.concurrent.RecursiveAction.exec(RecursiveAction.java:194)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:507)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1491)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:2073)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:2035)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:187)
```

Seems like a requirement for jvm flags comes from chronicle-hft.

File: plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java (hunk 6 above)
Comment (wendigo) [reply plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotLatestNoGrpcConnectorSmokeTest.java:31]: Sure. Let me drop it.

File: plugin/trino-pinot/pom.xml (hunk 5 above)
Comment (wendigo) [reply plugin/trino-pinot/pom.xml:27]: I think that it is required but just one flag is needed. I'll push in a second

Comment (wendigo) [reply plugin/trino-pinot/pom.xml:27]: `PinotDataBuffer` is used on the product path

Comment (wendigo) [reply plugin/trino-pinot/pom.xml]: PTAL now.

Comment (wendigo) [reply plugin/trino-pinot/pom.xml]: Only `--add-opens=java.base/java.lang.reflect=ALL-UNNAMED` is needed

File: plugin/trino-pinot/pom.xml [hunk 7]
```diff
@@ -14,15 +14,18 @@
 
     <properties>
         <air.main.basedir>${project.parent.basedir}</air.main.basedir>
-        <dep.pinot.version>0.12.1</dep.pinot.version>
+        <dep.pinot.version>1.1.0</dep.pinot.version>
+        <!-- additional JVM flags required by chronicle-hft which is used by pinot-segment spi -->
+        <air.test.jvm.additional-arguments>${air.test.jvm.additional-arguments.default}
+            --add-opens=java.base/java.lang.reflect=ALL-UNNAMED</air.test.jvm.additional-arguments>
```
Comment (ebyhr) [inline plugin/trino-pinot/pom.xml:20]: We should document this option in pinot.md. Follow-up is fine.

**Review by ebyhr — APPROVED** [review]

**Pull request merged**